vnext/templates/cpp-app/windows/MyApp/MyApp.cpp

```diff
@@ -0,0 +1,284 @@
+    host.InstanceSettings().UseFastRefresh(m_fastRefreshEnabled);
+    host.InstanceSettings().DebuggerPort(m_debuggerPort);
+    host.InstanceSettings().UseDeveloperSupport(true);
+
+    host.PackageProviders().Append(winrt::make<CompReactPackageProvider>());
+    winrt::Microsoft::ReactNative::ReactCoreInjection::SetTopLevelWindowId(
+        host.InstanceSettings().Properties(), reinterpret_cast<uint64_t>(hwnd));
+
+    // Nudge the ReactNativeHost to create the instance and wrapping context
+    host.ReloadInstance();
+
+    winrt::Microsoft::ReactNative::ReactViewOptions viewOptions;
+    viewOptions.ComponentName(appName);
+    m_CompositionHwndHost.ReactViewHost(
+        winrt::Microsoft::ReactNative::ReactCoreInjection::MakeViewHost(host, viewOptions));
+
+    auto windowData = WindowData::GetFromWindow(hwnd);
+    if (!windowData->m_windowInited) {
+      m_CompositionHwndHost.Initialize((uint64_t)(hwnd));
+      windowData->m_windowInited = true;
+    }
+    return 0;
+  }
+
+  LRESULT TranslateMessage(UINT message, WPARAM wparam, LPARAM lparam) noexcept {
+    if (m_CompositionHwndHost) {
+      return static_cast<LRESULT>(m_CompositionHwndHost.TranslateMessage(message, wparam, lparam));
+    }
+    return 0;
+  }
+};
+
+extern "C" IMAGE_DOS_HEADER __ImageBase;
+HINSTANCE WindowData::s_instance = reinterpret_cast<HINSTANCE>(&__ImageBase);
+
+LRESULT CALLBACK WndProc(HWND hWnd, UINT message, WPARAM wParam, LPARAM lParam) {
+  auto windowData = WindowData::GetFromWindow(hWnd);
+  if (windowData) {
+    auto result = WindowData::GetFromWindow(hWnd)->TranslateMessage(message, wParam, lParam);
+    if (result)
+      return result;
+  }
+
+  switch (message) {
+    case WM_DESTROY: {
+      delete WindowData::GetFromWindow(hWnd);
+      SetProp(hWnd, WindowDataProperty, 0);
+      PostQuitMessage(0);
+      return 0;
+    }
+    case WM_NCCREATE: {
+      auto cs = reinterpret_cast<CREATESTRUCT *>(lParam);
+      auto windowData = static_cast<WindowData *>(cs->lpCreateParams);
+      WINRT_ASSERT(windowData);
+      SetProp(hWnd, WindowDataProperty, reinterpret_cast<HANDLE>(windowData));
+      break;
+    }
+    case WM_GETOBJECT: {
+      {
+        auto windowData = WindowData::GetFromWindow(hWnd);
+        if (windowData == nullptr || !windowData->m_windowInited)
+          break;
+
+        auto hwndHost = windowData->m_CompositionHwndHost;
+        winrt::com_ptr<IRawElementProviderSimple> spReps;
+        if (!hwndHost.UiaProvider().try_as(spReps)) {
+          break;
+        }
+        LRESULT lResult = UiaReturnRawElementProvider(hWnd, wParam, lParam, spReps.get());
+        return lResult;
+      }
+    }
+  }
+
+  return DefWindowProc(hWnd, message, wParam, lParam);
+}
+
+int RunRNTester(int showCmd) {
+  auto windowData = std::make_unique<WindowData>(winrt::Microsoft::ReactNative::CompositionHwndHost());
+  HWND hwnd = CreateWindow(
+      c_windowClassName,
+      appName,
```

Comment:
appName as the window title generally makes for a poor window title. Could make it more obvious as a point of customization to folks?
```
constexpr PCWSTR windowTitle = appName
```

Reply:
This will most likely be subsumed by a larger refactor, but tracking bug here: https://github.com/microsoft/react-native-windows/issues/12180